chips/e310x/src/chip.rs

```diff
@@ -167,7 +169,18 @@ unsafe fn handle_interrupt(intr: mcause::Interrupt) {
             CSR.mie.modify(mie::mtimer::CLEAR);
         }
         mcause::Interrupt::MachineExternal => {
+            // We recieved an interrupt, disable interrupts while we handle them
             CSR.mie.modify(mie::mext::CLEAR);
+
+            // Claim the interrupt, unwrap() as we know an interrupt exists
```

Comment:
Would it make sense to merge the arrays, rather than claim the next?

I.e., is it possible to add multiple interrupts at once to the saved set, based on which are pending? I don't think this is a big deal, just wondering.

Reply:
Do you mean loop over `PLIC.next_pending()` until there are no more pending interrupts?

Reply:
I'd been thinking more of merging the pending bits in the interrupt pending bits block with this software state using a logical OR. But when I suggested this, I didn't realize that the block is 32 registers (128 bytes) long. The benefit from reducing the number of interrupts handled when there are multiple pending would in most cases be small compared to the cost of having to merge 32 words on each interrupt.

However, I do think that looping over PLIC.next_pending() a few times would be good; if there happen to be a few pending interrupts, amortizing the handling cost across them would be a big savings.

Reply:
Done! We now loop over `next_pending()` in the interrupt handler.